client.go

```diff
@@ -37,7 +37,7 @@ var (
 		KeepAlive: 30 * time.Second,
 	}
 	defaultNameTTL  = 5 * time.Minute
-	defaultResolver = resolver.NewDNSResolver(net.DefaultResolver, "ip", defaultNameTTL, resolver.PreferIPv4)
+	defaultResolver = resolver.NewDNSResolver(net.DefaultResolver, resolver.PreferIPv4, defaultNameTTL)
```

Comment:
I think we want this default to be controllable via env var. Maybe something like `HTTPLB_DEFAULT_ADDRESS_FAMILY_POLICY`. We can document the exact values the env var needs to have either in `NewClient` or maybe it would be better in `doc.go` where it talks about default behavior.

Reply:
Hmmm... To be honest, I'm not sold on this. There are exceptions where it may be useful (a la `GODEBUG`) but in the general case I see this as a code smell. It essentially adds an implicit "impure" configuration knobs that skip the application's own configuration mechanisms. If an application wants to create a "default" resolver with one setting changed, they have to replicate this behavior too.

If applications want this behavior to be controlled by environment variable, I think they ought to do it themselves.

If you insist I can add it, but is there really a good reason why reading and parsing environment variables is something `httplb` itself should do?

Reply:
> skip the application's own configuration mechanisms

It would only be for the default, where the application has not configured anything (which will typically be unintentional, either an inadvertent omission or lack of foresight). If the application _does_ configure it, the env var would not override.

> If applications want this behavior to be controlled by environment variable, I think they ought to do it themselves.

I guess my worry is that this would never happen in practice. If they are adding code of this sort at all, then they are thinking about this and likely configuring things intentionally. More likely is that this isn't thought about or configured at all because the defaults "just worked" on day one. And then on day two, when some vendor pushes a DNS change, things break (such as in an IPv6 single-stack setup, where DNS is updated to suddenly start returning an A record). I guess I'm just used to providing knobs for certain scenarios that don't changing code, re-building, and re-deploying, as a dev ops instinct. 🤷 

I guess we can leave it out. But this is the only behavior where httplb will observably differ from net/http and could do so in a way that prevents successful client->server communication, so it seemed useful. To be honest, I don't place a lot of value on "purity", especially if it potentially compromises usability or developer experience.

Anyhow, if you really don't like it, leave it out. But we definitely _do_ need to update the docs about the change in default behavior.

Reply:
I think I'll hold off on adding this. I can see the operational value of a feature like this, but in practice I think that it is least surprising to users if a library has one and only one way of configuring things, through the API, rather than environment variables. If we do want to make this sort of thing configurable through environment variables, it'd probably be worth considering _other_ things that might be worth exposing as environment variables and/or flags, possibly as a side-effect of importing an optional package, kind of like how some C++ libraries will have packages you can link to to get absl flags. I can see some arguments against this, too (it wouldn't help for people who didn't explicitly think of this, which is arguably the people who have the most to gain from the built-in ability to change a configuration knob at runtime) but ultimately I think that the better way to resolve this potential need is to make our IPv6 fallback work better in the future, so that users don't really have to think about it much more than they would for `net.Dial` and `http.DefaultClient`.